Description: Simple spelling fix
Author: Alberto Gonzalez Iniesta <agi@inittab.org>
Index: openvpn/src/openvpn/occ.c
===================================================================
--- openvpn.orig/src/openvpn/occ.c	2014-05-14 12:58:59.805184504 +0200
+++ openvpn/src/openvpn/occ.c	2014-05-14 12:58:59.805184504 +0200
@@ -379,7 +379,7 @@
 	      && c->c2.max_send_size_local > TUN_MTU_MIN
 	      && (c->c2.max_recv_size_remote < c->c2.max_send_size_local
 		  || c->c2.max_recv_size_local < c->c2.max_send_size_remote))
-	    msg (M_INFO, "NOTE: This connection is unable to accomodate a UDP packet size of %d. Consider using --fragment or --mssfix options as a workaround.",
+	    msg (M_INFO, "NOTE: This connection is unable to accommodate a UDP packet size of %d. Consider using --fragment or --mssfix options as a workaround.",
 		 c->c2.max_send_size_local);
 	}
       event_timeout_clear (&c->c2.occ_mtu_load_test_interval);
