Description: Improve kFreeBSD support
Author: Gonéri Le Bouder <goneri@rulezlan.org>
Bug-Debian: http://bugs.debian.org/626062
Index: openvpn/src/openvpn/route.c
===================================================================
--- openvpn.orig/src/openvpn/route.c	2016-01-20 12:02:41.441269913 +0100
+++ openvpn/src/openvpn/route.c	2016-01-20 12:02:41.437269956 +0100
@@ -1421,7 +1421,7 @@
   argv_msg (D_ROUTE, &argv);
   status = openvpn_execve_check (&argv, es, 0, "ERROR: Solaris route add command failed");
 
-#elif defined(TARGET_FREEBSD)
+#elif defined(TARGET_FREEBSD)||defined(__FreeBSD_kernel__)
 
   argv_printf (&argv, "%s add",
 		ROUTE_PATH);
Index: openvpn/src/openvpn/tun.c
===================================================================
--- openvpn.orig/src/openvpn/tun.c	2016-01-20 12:02:41.441269913 +0100
+++ openvpn/src/openvpn/tun.c	2016-01-20 12:02:41.437269956 +0100
@@ -1124,7 +1124,7 @@
 	  add_route_connected_v6_net(tt, es);
 	}
 
-#elif defined(TARGET_FREEBSD)||defined(TARGET_DRAGONFLY)
+#elif defined(TARGET_FREEBSD)||defined(TARGET_DRAGONFLY)||defined(__FreeBSD_kernel__)
 
       /* example: ifconfig tun2 10.2.0.2 10.2.0.1 mtu 1450 netmask 255.255.255.255 up */
       if (tun)
