Index: openvpn-2.1_rc15/options.c
===================================================================
--- openvpn-2.1_rc15.orig/options.c	2009-04-30 12:58:46.952616319 +0200
+++ openvpn-2.1_rc15/options.c	2009-04-30 12:58:50.352666598 +0200
@@ -769,8 +769,8 @@
   setenv_str_i (es, "proto", proto2ascii (e->proto, false), i);
   setenv_str_i (es, "local", e->local, i);
   setenv_int_i (es, "local_port", e->local_port, i);
-  setenv_str_i (es, "remote", e->local, i);
-  setenv_int_i (es, "remote_port", e->local_port, i);
+  setenv_str_i (es, "remote", e->remote, i);
+  setenv_int_i (es, "remote_port", e->remote_port, i);
 
 #ifdef ENABLE_HTTP_PROXY
   if (e->http_proxy_options)
