--- openssl-1.0.0j/Configurations/unix-Makefile.tmpl.orig	2019-02-20 08:43:59.518452803 +0000
+++ openssl-1.0.0j/Configurations/unix-Makefile.tmpl	2019-02-20 09:11:32.730452803 +0000
@@ -496,6 +496,7 @@
 	@$(PERL) $(SRCDIR)/util/mkdir-p.pl $(DESTDIR)$(INSTALLTOP)/$(LIBDIR)
 	@ : {- output_on() if windowsdll(); "" -}
 	@echo "*** Installing runtime files"
+	@$(PERL) $(SRCDIR)/util/mkdir-p.pl $(DESTDIR)$(INSTALLTOP)/bin
 	@set -e; for s in dummy $(INSTALL_SHLIBS); do \
 		if [ "$$s" = "dummy" ]; then continue; fi; \
 		fn=`basename $$s`; \
