Index: openvpn-2.1.3/ssl.h
===================================================================
--- openvpn-2.1.3.orig/ssl.h	2011-03-11 12:35:32.000000000 +0100
+++ openvpn-2.1.3/ssl.h	2011-03-11 12:59:08.883318001 +0100
@@ -378,8 +378,8 @@
 
   struct buffer_list *paybuf;
 
-  int n_bytes;			 /* how many bytes sent/recvd since last key exchange */
-  int n_packets;		 /* how many packets sent/recvd since last key exchange */
+  counter_type n_bytes;		 /* how many bytes sent/recvd since last key exchange */
+  counter_type n_packets;	 /* how many packets sent/recvd since last key exchange */
 
   /*
    * If bad username/password, TLS connection will come up but 'authenticated' will be false.
